### PR TITLE
Fix #7104 - tf.initializers.<random | glorot | he | leCunn>Uniform() ignores seed argument

### LINKED ISSUE #7104: tf.initializers.<random | glorot | he | leCunn>Uniform() ignores seed argument
**System information**
- Have I written custom code: NO
- OS Platform and Distribution: macOS Monterey 12.5.1
- Mobile device: NO
- TensorFlow.js installed: [npm](https://www.npmjs.com/package/@tensorflow/tfjs)
- TensorFlow.js version: [3.17](https://www.npmjs.com/package/@tensorflow/tfjs/v/3.17.0) but it replicates on all versions
- ~~Browser~~ version: Node.js v16.15.1 but it replicates no matter Browser vs Node or their version
- Tensorflow.js Converter Version: NO (not involved)

**Describe the current behavior**

If you use any of [randomUniform](https://js.tensorflow.org/api/latest/#initializers.randomUniform), [glorotUniform](https://js.tensorflow.org/api/latest/#initializers.glorotUniform), [heUniform](https://js.tensorflow.org/api/latest/#initializers.heUniform), [leCunUniform](https://js.tensorflow.org/api/latest/#initializers.leCunUniform) with `seed` argument it has no effect.

Every time you start a new training you get a different set of weights, aka the generated random sequence is not deterministic.

**Describe the expected behavior**

I expect that if you use the same `seed` for all the initializers then every time you start a new training you get the same set of weights, aka the generated random sequence to be deterministic.

**Standalone code to reproduce the issue**

Every time you will run the code below you will get a different sequence of values for the weight:

```typescript
import * as tf from '@tensorflow/tfjs-node';

class DummyCustomLayer extends tf.layers.Layer {

    numChannels: number;

    someKernel?: tf.LayerVariable;

    constructor({}) {
        super({ name: 'DummyCustomLayer' });

        this.numChannels = 80;
    }

    build(inputShape: tf.Shape) {
        super.build(inputShape);

        this.someKernel = this.addWeight(
            'some_kernel',
            [inputShape[inputShape.length - 1], this.numChannels],
            'float32',
            tf.initializers.glorotUniform({ seed: 42 })
        );
    }

    call(input: tf.Tensor) {

        tf.print(this.someKernel!.read(), true); // debug purpose

        return tf.matMul(input, this.someKernel!.read());
    }

    computeOutputShape(inputShape: tf.Shape) {
        return [...inputShape.slice(0, -1), this.numChannels]
    }
}

const example = new DummyCustomLayer({});
const result = example.apply(tf.zeros([10, 20]));
```

### PR DESCRIPTION
**Pull request by adrian-branescu**
Fix #7104 - tf.initializers.<random | glorot | he | leCunn>Uniform() ignores seed argument & add tests that replicated the issue, fix wrong serialization name registered for LeCunUniform initializer class

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7108)

### CODE REVIEW

**Review by pyu10055 — APPROVED** [review]
thank you for the fix.

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7108#-:-NI3N1UR1qwHH3cY3rYo:b-fubd5x)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @mattsoulanille)

**Review by mattsoulanille — APPROVED** [review]
Thanks for the fix! LGTM.
